lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -752,12 +770,68 @@ selectLatestCheckpoint wid = fmap entityVal <$>
 selectUTxO
     :: Checkpoint
     -> SqlPersistT IO [UTxO]
-selectUTxO (Checkpoint wid sl _parent _bh) = fmap entityVal <$>
+selectUTxO (Checkpoint wid _sl _parent _bh) = fmap entityVal <$>
     selectList
         [ UtxoWalletId ==. wid
-        , UtxoSlot ==. sl
+        , UtxoSlotSpent ==. Nothing
         ] []
 
+insertWhenNew
+    :: W.WalletId
+    -> UTxO
+    -> SqlPersistT IO ()
+insertWhenNew wid utxo@(UTxO _ _ _ inpId inpIx outAddr outAmt) = do
+    utxos <- selectList [ UtxoWalletId ==. wid
+                        , UtxoInputId ==. inpId
+                        , UtxoInputIndex ==. inpIx
+                        , UtxoOutputAddress ==. outAddr
+                        , UtxoOutputCoin ==. outAmt
+                        ] []
+    when (null utxos) $ insert_ utxo
```

Comment:
can be removed as hinted before, unless I am missing something :thinking: ?

Reply:
well, I want to insert only new utxo, if the utxo was present in previous slots I do not want to do anything

Reply:
Leave that to the database: `INSERT INTO utxos <values> WHERE NOT EXISTS (SELECT * FROM utxos WHERE id IN <utxos_ids>)`

Reply:
Unfortunately I don't think that we can express subqueries with plain persistent functions. Anyway I have removed this function.